packages/fx-core/src/core/tools.ts

```diff
@@ -118,6 +119,24 @@ export function isValidProject(workspacePath?: string): boolean {
   }
 }
 
+export function getActiveEnv(projectRoot: string): string | undefined {
+  if (!isMultiEnvEnabled()) {
+    return "default";
+  }
+  try {
+    if (isValidProject(projectRoot)) {
+      const settingsJsonPath = path.join(
+        projectRoot,
+        `.${ConfigFolderName}/${InputConfigsFolderName}/${ProjectSettingsFileName}`
+      );
+      const settingsJson = JSON.parse(fs.readFileSync(settingsJsonPath, "utf8"));
```

Comment:
async method would be always preferred: `fs.readFileSync`

Reply:
Currently, the config related util functions are all sync functions. We need to refactor other util functions that use this function. It makes sense to change them to async but the refactor effort kind of big, so I won't cover in this pr.

Reply:
Sure thing.